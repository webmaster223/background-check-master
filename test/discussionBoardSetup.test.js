const discussionBoardSetup = require('../lib/discussionBoardSetup')

/* -- Mocks -- */
const createDiscussionRepo = jest.fn()
const addCollaboratorToRepo = jest.fn()
/* ^- Mocks -^ */

const discussionBoardSetupInstance = discussionBoardSetup({
  dependencies: {
    createDiscussionRepo,
    addCollaboratorToRepo
  }
})

test('discussionBoardSetup is working', async () => {
  const context = {
    payload: {
      installation: { account: { login: 'itaditya' } },
      sender: { login: 'itaditya' }
    },
    log: console.log
  }
  await discussionBoardSetupInstance(context)
  expect(createDiscussionRepo.mock.calls).toMatchSnapshot()
  expect(addCollaboratorToRepo.mock.calls).toMatchSnapshot()
})
