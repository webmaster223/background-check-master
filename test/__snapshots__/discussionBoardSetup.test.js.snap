// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`discussionBoardSetup is working 1`] = `
Array [
  Array [
    undefined,
    Object {
      "appInstallerName": "itaditya",
    },
  ],
]
`;

exports[`discussionBoardSetup is working 2`] = `
Array [
  Array [
    undefined,
    Object {
      "appInstallerName": "itaditya",
      "username": "itaditya",
    },
  ],
]
`;
